assets/src/edit-story/components/carousel/secondaryMenu.js

```diff
@@ -97,14 +109,23 @@ function SecondaryMenu() {
   useEffect(() => {
     if (isChecklistOpen) {
       closeHelpCenter();
+      closeKeyboardShortcutsMenu();
     }
-  }, [closeHelpCenter, isChecklistOpen]);
+  }, [closeHelpCenter, closeKeyboardShortcutsMenu, isChecklistOpen]);
 
   useEffect(() => {
     if (isHelpCenterOpen) {
       closeChecklist();
+      closeKeyboardShortcutsMenu();
+    }
+  }, [closeChecklist, closeKeyboardShortcutsMenu, isHelpCenterOpen]);
+
+  useEffect(() => {
+    if (isKeyboardShortcutsMenuOpen) {
+      closeChecklist();
+      closeHelpCenter();
     }
-  }, [closeChecklist, isHelpCenterOpen]);
+  }, [closeChecklist, closeHelpCenter, isKeyboardShortcutsMenuOpen]);
```

Comment:
This looks like it'd lend itself to a singular popup provider pattern where the state can only be one popup at time. something like:
```
const [openPopup, setOpenPopup] = useState('checklist' | 'helper' | 'keyboard shortcuts' | undefined);
```
then we wouldn't have to manually close all the other ones, would just be a natural byproduct of the state structure. Anyway, something to consider for another time if we come across a similar popup rich feature in future designs.

I think the implementation you have here is easy to follow and the least impactful change to get the desired behavior given the existing setup tho, so I'm diggin it

Reply:
that's a great idea. the refactor of the popup that we're hooking into for the checklist from the helper is still up for grabs for this next sprint - could be a good place to try that out.